conda_gitenv/deploy.py

```diff
@@ -42,63 +44,110 @@ def tags_by_env(repo):
     return tags
 
 
-def deploy_tag(repo, tag_name, target):
+def deploy_tag(repo, tag_name, target, api_user=None, api_key=None,
+               mirror=None):
     tag = repo.tags[tag_name]
     # Checkout the tag in a detached head form.
     repo.head.reference = tag.commit
     repo.head.reset(working_tree=True)
 
-    # Pull out the environment name from the form "env-<env_name>-<deployed_name>".
+    # Parse tag_name with form "env-<env_name>-<deployed_name>".
     env_name = tag_name.split('-')[1]
     deployed_name = tag_name.split('-', 2)[2]
 
     manifest_fname = os.path.join(repo.working_dir, 'env.manifest')
     if not os.path.exists(manifest_fname):
-        raise ValueError("The tag '{}' doesn't have a manifested environment.".format(tag_name))
+        msg = "The tag '{}' doesn't have a manifested environment."
+        raise ValueError(msg.format(tag_name))
     with open(manifest_fname, 'r') as fh:
         manifest = sorted(line.strip().split('\t') for line in fh)
 
+    # Replace the channel URL with the mirror URL for each package
+    # entry specified in the manifest.
+    if mirror is not None:
+        manifest = [[mirror, pkg] for channel, pkg in manifest]
```

Comment:
Here you replace every channel, regardless of what they are, even if they differ. 
Which means that if you had a manifest that was something like:
```
https://mymachine:1234/my_channel/linux-64    python
https://mymachine:1234/my_channel/no-arch     numpy
````
it requires python and numpy to live in the same mirror, as your manifest now looks like:
```
/path/to/mirror    python
/path/to/mirror    numpy
````

Reply:
@lbdreyer Yup, nice spot. It should instead replace with `/path/to/mirror/linux-64` and `/path/to/mirror/noarch` etc ... fix in-bound :+1:

Reply:
@lbdreyer Interesting that you used a directory path `/path/to/mirror` instead of a URL `http://path/to/mirror` in your example above. I've been testing with a URL, but I'll just confirm that it works with a FS directory as well...

Reply:
Hmmm it doesn't work when the provided `mirror` is a directory, so it might be worthwhile restricting this here in this PR, and support that later on, if it's required.